forum/views.py

```diff
@@ -87,7 +88,12 @@ def post(self, request, *args, **kwargs):
         if not request.user.is_authenticated:
             self.handle_no_permission()
         else:
-            return super().post(request, *args, **kwargs)
+            thread = get_object_or_404(Thread, id=self.kwargs['thread'])
```

Comment:
does the following if statement handle a 404 properly?

Reply:
Yes, as a 404 is an exception in django, execution stops at this line if the object isn't found